File: traversals/src/test/scala/io/shiftleft/overflowdb/testdomains/gratefuldead/GratefulDead.scala
package io.shiftleft.overflowdb.testdomains.gratefuldead

import java.util

import io.shiftleft.overflowdb.{OdbConfig, OdbGraph}
import org.apache.tinkerpop.gremlin.structure.io.IoCore

object GratefulDead {
  def newGraph: OdbGraph = newGraph(OdbConfig.withoutOverflow)

  def newGraph(config: OdbConfig): OdbGraph =
    OdbGraph.open(config,
      util.Arrays.asList(Song.factory, Artist.factory),
      util.Arrays.asList(FollowedBy.factory, SungBy.factory, WrittenBy.factory))

  def newGraphWithData: OdbGraph = {
    val graph = newGraph
    loadData(graph)
    graph
  }

  def loadData(graph: OdbGraph): Unit =
    graph.io(IoCore.graphml).readGraph("src/test/resources/grateful-dead.xml")

  def traversal(graph: OdbGraph) = new GratefulDeadTraversalSource(graph)
}

File: core/src/test/java/overflowdb/testdomains/gratefuldead/GratefulDead.java
package overflowdb.testdomains.gratefuldead;

import overflowdb.Config;
import overflowdb.Graph;

import java.util.Arrays;

public class GratefulDead {
  public static Graph newGraph() {
    return newGraph(Config.withoutOverflow());
  }

  public static Graph newGraph(Config config) {
    return Graph.open(
        config,
        Arrays.asList(Song.factory, Artist.factory),
        Arrays.asList(FollowedBy.factory, SungBy.factory, WrittenBy.factory)
    );
  }

}
